### PR TITLE
Test/plugins endpoint

### PR DESCRIPTION
**Pull request by suzannemaquelin**
I pulled business logic out of RequiredInfo in the same way Lars did for devices endpoints. Made test for RequiredInfo similar to test of PluginManager. Also, changed some documentation according to Luc's requests

### CODE REVIEW

File: source/endpoints/devices/Device.py [hunk 1]
```diff
@@ -25,11 +24,11 @@ class Device(Resource):
     @namespace.marshal_with(device)
     def get(self, device_id):
         """ Fetch a device given its identifier """
-        return device_database.get_device(device_id)
+        return business_logic_device.get_device_from_database_by_id(device_id)
```
Comment (szijerveld) [inline source/endpoints/devices/Device.py]: In my opinion the from_database is a little to much. 
It is up to the business logic to decide if it is retrieved form a database of from somewhere else.

File: source/endpoints/devices/Device.py [hunk 2]
```diff
@@ -1,9 +1,8 @@
 from flask_restplus import Resource
 from flask_restplus import fields
 
-from endpoints.devices import namespace
+from endpoints.devices import namespace, business_logic_device
```
Comment (szijerveld) [inline source/endpoints/devices/Device.py]: I think that the business logic should be in a seperate module and not in he endpoint module

File: source/endpoints/devices/Devices.py [hunk 3]
```diff
@@ -15,22 +15,19 @@
...
         """ Post a new device """
-        required_info = namespace.apis[0].payload["required_info"]
-        new_device_info = eval(required_info)
-        name = new_device_info.pop("name", None)
-        class_plugin = plugin_manager.get_implementation_of(new_device_info)
-        class_plugin.name = name
-        device_database.add_device(class_plugin)
-        class_plugin.setup()
+        json_required_info = namespace.apis[0].payload["required_info"]
+        plugin = business_logic_devices.create_new_device_from_json_input(json_required_info)
```
Comment (szijerveld) [inline source/endpoints/devices/Devices.py]: Now there is still some business logic in the endpoint with two calls to the business logic class,

File: source/endpoints/plugins/RequiredInfo.py [hunk 4]
```diff
@@ -35,6 +35,4 @@ class RequiredInfo(Resource):
     @namespace.doc("get_activator")
     def get(self, organization, plugin_name):
         """ Fetch the required information of a plugin based on its organization and name """
-        required_info = plugin_manager.get_required_info_of(organization, plugin_name)
-        required_info["name"] = "default"
-        return required_info
\ No newline at end of file
+        return business_logic_required_info.get_required_info_by_organization_and_plugin_name(organization, plugin_name)
```
Comment (szijerveld) [inline source/endpoints/plugins/RequiredInfo.py]: This line is rather hard to read.

File: source/tests/endpoints/plugins/test_requiredInfo.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import unittest
+
+from endpoints.plugins.business_logic.RequiredInfo import BusinessLogicRequiredInfo
+from model.PluginManager import PluginManager
+from plugins.mock.lock.Lock import Lock
+
+
+class TestEndpointRequiredInfo(unittest.TestCase):
+    """
+    A test for the endpoint RequiredInfo.
```
Comment (szijerveld) [inline source/tests/endpoints/plugins/test_requiredInfo.py]: This test does not test the endpoint itself does it it tests the business logic module.

File: source/endpoints/plugins/business_logic/RequiredInfo.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+class BusinessLogicRequiredInfo:
```
Comment (szijerveld) [inline source/endpoints/plugins/business_logic/RequiredInfo.py:1]: To be honest I do not really see the benefits of having a wrapper around the plugin manager as almost all the this is does is calling the plugin_manager.

**Review by szijerveld — CHANGES_REQUESTED** [review]

File: source/endpoints/devices/Device.py (hunk 2 above)
Comment (szijerveld) [reply source/endpoints/devices/Device.py]: We might want to fix this next sprint though.
So you can leave as such.

File: source/endpoints/plugins/business_logic/RequiredInfo.py (hunk 6 above)
Comment (suzannemaquelin) [reply source/endpoints/plugins/business_logic/RequiredInfo.py:1]: It's in line with device endpoints

Comment (szijerveld) [reply source/endpoints/plugins/business_logic/RequiredInfo.py:1]: Yeah that's true.

**Review by szijerveld — APPROVED** [review]

**Pull request merged**